fastly/image_optimizer_default_settings.go

```diff
@@ -0,0 +1,187 @@
+package fastly
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// ResizeFilter is a base for the different ResizeFilter variants.
+type ResizeFilter int64
+
+func (r ResizeFilter) String() string {
+	switch r {
+	case Lanczos3:
+		return "lanczos3"
+	case Lanczos2:
+		return "lanczos2"
+	case Bicubic:
+		return "bicubic"
+	case Bilinear:
+		return "bilinear"
+	case Nearest:
+		return "nearest"
+	}
+	return "lanczos3" // default
+}
+
+func (r ResizeFilter) MarshalJSON() ([]byte, error) {
+	return json.Marshal(r.String())
+}
+
+const (
+	// A Lanczos filter with a kernel size of 3. Lanczos filters can detect edges and linear features within an image, providing the best possible reconstruction.
+	Lanczos3 ResizeFilter = iota
+	// A Lanczos filter with a kernel size of 2.
+	Lanczos2
+	// A filter using an average of a 4x4 environment of pixels, weighing the innermost pixels higher.
+	Bicubic
+	// A filter using an average of a 2x2 environment of pixels.
+	Bilinear
+	// A filter using the value of nearby translated pixel values. Preserves hard edges.
+	Nearest
+)
+
+// JpegType is a base for different JpegType variants
+type JpegType int64
+
+func (r JpegType) String() string {
+	switch r {
+	case Auto:
+		return "auto"
+	case Baseline:
+		return "baseline"
+	case Progressive:
+		return "progressive"
+	}
+	return "auto" // default
+}
+
+func (r JpegType) MarshalJSON() ([]byte, error) {
+	return json.Marshal(r.String())
+}
+
+const (
+	// Match the input JPEG type, or baseline if transforming from a non-JPEG input.
+	Auto JpegType = iota
+	// Output baseline JPEG images
+	Baseline
+	// Output progressive JPEG images
+	Progressive
```

Comment:
These constants should be prefixed with `ImageOpto` as they will be globally available across the whole go-fastly package and in the future there might be other features/APIs that have things related to 'auto' or 'progressive' etc, so we need to be able to distinguish these types as being related to ImageOpto.

Reply:
Right, of course! I do have a question - now that I'm looking for it, I see the other files doing this have the type name as a prefix to each constant - `ProductUndefined` for example in product_enablement.

Do you think `ResizeFilterAuto`, etc. (and `JpegTypeAuto`, etc.) could make sense, or would that still be too unspecific?

Reply:
sounds good!

Reply:
Should be fixed!